### PR TITLE
Tackles issue #67

### PR DESCRIPTION
**Pull request by brunnerant**
See issue #67 for a description of what this PR aims at ;).

### CODE REVIEW

**Comment by brunnerant** [comment]
The tests fail because I have not yet updated them to reflect the changes that I have made to `ListEditView`.

**Comment by brunnerant** [comment]
This PR is good to go. There was a problem with the coverage of Cosme's PR. It may be related to the fact the he works on a fork of the repo.... Anyway, I refactored what he wrote and wrote some more thorough tests, so it is ok now.

**Comment by brunnerant** [comment]
Maybe a few comments on what I did and what work would need to follow. I have added the list edit view to the home activity, which I have refactored into a nice class `ListEditView`. This class can be listened through a `ListEditAdapter`, so that all the events can be processed properly.
The thing that we will have to handle better in the next PRs is to dispatch the `HomeQuizListFragment` whether the user is an editor or a participant. Right now, all the users can edit the list of quizzes, which doesn't make sense.

File: app/src/main/java/ch/epfl/qedit/view/util/ConfirmDialog.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        Bundle args = getArguments();
+        message = args.getString("message");
+        listener = (ConfirmationListener) args.getSerializable("listener");
+
+        return new AlertDialog.Builder(getActivity())
```
Comment (nbaehler) [inline app/src/main/java/ch/epfl/qedit/view/util/ConfirmDialog.java:55]: Personally I have a trauma from getActivity() ... ;) Maybe it is not justified but I'm suggesting to use requireActivity()

File: app/src/main/java/ch/epfl/qedit/view/util/EditTextDialog.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        Bundle args = getArguments();
+        message = args.getString("message");
+        listener = (SubmissionListener) args.getSerializable("listener");
+
+        dialog =
+                new AlertDialog.Builder(getActivity())
```
Comment (nbaehler) [inline app/src/main/java/ch/epfl/qedit/view/util/EditTextDialog.java:126]: RequireActivity() would possibly be safer.

File: app/src/main/java/ch/epfl/qedit/view/util/ConfirmDialog.java (hunk 1 above)
Comment (brunnerant) [reply app/src/main/java/ch/epfl/qedit/view/util/ConfirmDialog.java:54]: Yes you are right, I will change that ;).

File: app/src/main/java/ch/epfl/qedit/view/util/EditTextDialog.java (hunk 2 above)
Comment (brunnerant) [reply app/src/main/java/ch/epfl/qedit/view/util/EditTextDialog.java:125]: Yes, I will also change that, that's a mistake from me -_-

**Comment by brunnerant** [comment]
Yes, you are totally right. But I didn't want to make that monolitic PR that creates conflicts everywhere, so I will do it in a future PR. Here is a list of things we could (or have to) do in the following PRs:
 - Dispatching between participants, editors and administrators. What I mean by that is that a participant should have a different `HomeActivity` because he shouldn't be able to modify quizzes. An editor should have the editing capabilities. Right now, every one can edit quizzes in the home activity.
 - When an editor clicks on the edit button, he should be redirected to the `EditActivity` (or whatever activity we created). Right now, he is redirected to the `QuizActivity`, because the `EditActivity` has not been made yet.
 - When a participant clicks on a quiz, he should be redirected to the `QuizActivity`.
We could make tasks out of those missing features, for the next sprint.

**Comment by nbaehler** [comment]
> Yes, you are totally right. But I didn't want to make that monolitic PR that creates conflicts everywhere, so I will do it in a future PR. Here is a list of things we could (or have to) do in the following PRs:
> 
>     * Dispatching between participants, editors and administrators. What I mean by that is that a participant should have a different `HomeActivity` because he shouldn't be able to modify quizzes. An editor should have the editing capabilities. Right now, every one can edit quizzes in the home activity.
> 
>     * When an editor clicks on the edit button, he should be redirected to the `EditActivity` (or whatever activity we created). Right now, he is redirected to the `QuizActivity`, because the `EditActivity` has not been made yet.
> 
>     * When a participant clicks on a quiz, he should be redirected to the `QuizActivity`.
>       We could make tasks out of those missing features, for the next sprint.

Yes I agree 100%.

**Pull request merged**